docs/exit_validation.md

```diff
@@ -87,3 +105,88 @@ All the general rules will apply in the MoreVP world.
+  - we gather and keep in `OMG.Watcher.ExitProcessor`'s persistent state the current state of in-flight txs and exits
+  - we periodically check the validity of this state, emitting events and allowing for actions as necessary
+  - we touch `OMG.API.State` only when the in-flight exit finalizes
+  - if some invalid IFE or piggyback runs unchallenged for too long, it should be a prompt to exit (ala `:unchallenged_exit` above)
+
+**[Diagram](https://docs.google.com/drawings/d/1UaAMZTJBbikTM0eFSbNM7ZVgi1HZrLTfwYCUvHGUaW0/edit?usp=sharing)** illustrates the flows described below (with the addition of handling on the Child Chain Server side for comparison).
+
+**NOTES on the diagram**:
+  - `ACTION` means both:
+      - an event delivered to the user
+      - the user should then be able to take some action on the root-chain
+
+There are two rather independent flows in play:
+
+### In-flight transaction tracking
+
+This flow is about the user employing the MoreVP exit game to secure their own broadcast transactions.
+
+This flow begins with transaction being broadcast via the Watcher.
+Every such transaction is remembered and dubbed "in-flight".
+Such in-flight txs, in case of any byzantine behavior by the child chain - one that would prompt an immediate exit, causes a prompt to start an in-flight exit.
+
+**TODO** - in the first implementation, tracking of in-flight transactions isn't done in the Watcher but punted on the user.
+The user should figure out how to exit which of their funds upon receiving a generic `byzantine_chain` event.
+
+### In-flight exit tracking
+
+This flow is about the user piggybacking (or not), challenging IFEs and responding to challenges related to IFEs that are **already started** on the root chain contract.
+
+#### new IFEs, competitors, piggybacks
+
+Any `InFlightExitStarted` should, after `eth_exit_finality_margin` cause the IFE to be tracked and tx added to something called `TxAppendix`.
+
+Any competitor published should `eth_exit_finality_margin`, be tracked and tx added to `TxAppendix`
+
+Any piggyback done should `eth_exit_finality_margin`, be taken into account.
+
+#### finalization
+
+Finalization behaves analogically to standard exits - finalization spends in `OMG.API.State` all utxos that actually exit **and existed in the child chain**.
+These spends that fail cause a byzantine chain condition (`unchallenged_exit`)
+
+#### "wait"
+
+🕐 Periodically (same as with standard exits), `ExitProcessor` should:
+  - find invalid piggybacks and do something
+  - find non-canonical transactions that are seen as canonical in the root chain contract
+  - find canonical transactions that are seen as non-canonical in the root chain contract
+
+#### Checking if I should piggyback
+
+Any IFE started might be one such that the user should piggyback onto.
+
+This happens when an input or an output of the in-flight transaction submitted mentions the user as owner.
```

Comment:
As above - outputs are only worth piggybacking if inputs are in non-withheld correct block.

Reply:
I'm not sure I follow.

Inputs come with inclusion proofs on the root chain contract, so even if they were withheld, they become "unwithheld" at the moment of `startInFlightExit`.

And even if they ended up being in an incorrect block, the drill should still be the same - IFE is justified for those interested, and they should all piggyback + a mass exit independently.